spec/ameba/rule/lint/useless_assign_spec.cr

```diff
@@ -475,6 +475,32 @@ module Ameba::Rule::Lint
         issue.location.to_s.should eq "source.cr:1:1"
         issue.message.should eq "Useless assignment to variable `foo`"
       end
+
+      it "doesn't report if top level variable defined inside class is referenced" do
+        s = Source.new %(
+          class A
+            foo : String? = "foo"
+          end
+
+          puts foo
+        )
+        subject.catch(s).should be_valid
+      end
+
+      it "doesn't report if top level variable assigned inside class and referenced" do
+        s = Source.new %(
+          class A
+            foo : String? = "foo"
+
+            bar do
+              foo = "bar"
+            end
+          end
+
+          puts foo
+        )
+        subject.catch(s).should be_valid
+      end
```

Comment:
```suggestion
      it "doesn't report if top level variable defined inside class is referenced" do
        expect_no_issues subject, <<-CRYSTAL
          class A
            foo : String? = "foo"

            puts foo
          end
          CRYSTAL
      end

      it "doesn't report if top level variable assigned inside class and referenced" do
        expect_no_issues subject, <<-CRYSTAL
          class A
            foo : String? = "foo"

            bar do
              foo = "bar"
            end

            puts foo
          end
          CRYSTAL
      end
```

Reply:
I would still use the old style for consistency in the file. Let's move to new spec matchers in a separate PR